pkg/vault/contracts/AggregatorRouter.sol

```diff
@@ -81,7 +114,13 @@ contract AggregatorRouter is IAggregatorRouter, RouterCommon {
     ) external nonReentrant onlyVault returns (uint256) {
         (uint256 amountCalculated, uint256 amountIn, uint256 amountOut) = _swapHook(params);
 
-        _vault.settle(params.tokenIn, amountIn);
+        if (params.kind == SwapKind.EXACT_OUT) {
+            _vault.settle(params.tokenIn, params.limit);
+            _sendTokenOut(params.sender, params.tokenIn, params.limit - amountIn, false);
```

Comment:
Maybe a small comment here: 
- we're returning the difference between max amount in (paid upfront) and actual amount in to the user
- `params.limit >= amountIn` at this point, validated by the vault

Reply:
Ok, I created this comment, but also added an error when the vault balance is smaller than the swap limit. It was reverting with "math underflow" inside the Vault